staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -494,6 +496,19 @@ func cleanDryRun(u *url.URL) string {
 	return strings.Join(utilsets.NewString(dryRun...).List(), ",")
 }
 
+func cleanUserAgent(ua string) string {
```

Comment:
@smarterclayton you mentioned that you have idea of how to bound the cardinality of user-agent labels. What did you have in mind beyond such heuristics?

Reply:
There are ways to do estimation where you don’t load the whole set (like a bloom filter but with counters), but that may be overkill here if we can simply point to audit logs

Reply:
Very simple compromise: collect a limited number here in-memory. First 10 get their own label. After than `"others"`.